Sources/KlaviyoSwiftExtension/KlaviyoExtension.swift

```diff
@@ -24,6 +30,32 @@ public enum KlaviyoExtensionSDK {
         bestAttemptContent: UNMutableNotificationContent,
         contentHandler: @escaping (UNNotificationContent) -> Void,
         fallbackMediaType: String = "jpeg") {
+        // handle badge setting from the push notification payload
```

Comment:
Can this be broken down like, to avoid some duplication. Need to figure out what to do when you can read from user defaults. 


```swift
      if let badgeConfig = bestAttemptContent.userInfo["badge_config"] as? String {
          var newBadgeValue: Int?
          
          guard let userDefaults = getAppGroupUserDefaults else {
              return // not sure what to do here?
          }
          
          switch badgeConfig {
          case KlaviyoBadgeConfig.incrementOne.rawValue:
              let currentBadgeCount = userDefaults.integer(forKey: "badgeCount")
              newBadgeValue = currentBadgeCount + 1
              
          default:
              if let badgeValue = bestAttemptContent.userInfo["badge_value"] as? Int {
                  newBadgeValue = badgeValue
              }
          }
          userDefaults.set(newBadgeValue, forKey: "badgeCount")
          bestAttemptContent.badge = newBadgeValue as? NSNumber
      }


    static var getAppGroupUserDefaults: UserDefaults? {
        guard let appGroup = Bundle.main.object(forInfoDictionaryKey: "Klaviyo_App_Group") as? String else {
            return nil
        }
        return UserDefaults(suiteName: appGroup)
    }
```

Reply:
I'm okay with just returning since if we're unable to read from UserDefaults it likely means the count would be going out of sync at that point. We already plan on having the `set_count` and `set_property` cases have their aps badge set to the `badge_value` so display wise they would still be okay, and we could likewise do this for any other future enum cases so they'd at least be displayed correctly.

Reply:
That's a good point. I'm aligned.